src/main/java/it/pagopa/pm/gateway/controller/PostePayPaymentTransactionsController.java

```diff
@@ -430,15 +430,26 @@ private ResponseEntity<PostePayAuthResponse> createPostePayAuthResponse(String c
 
     private PostePayPollingResponse createPollingResponse(String requestId, PaymentRequestEntity entity) {
         PostePayPollingResponse response = new PostePayPollingResponse();
+        String guid = entity.getGuid();
+        String status = entity.getStatus();
         Boolean outcome = entity.getAuthorizationOutcome();
-        OutcomeEnum authorizationOutcome = Objects.isNull(outcome) ? null : outcome ? OK : KO;
         String urlRedirect = entity.getAuthorizationUrl();
+        OutcomeEnum authorizationOutcome = Objects.isNull(outcome) ? null : outcome ? OK : KO;
         response.setUrlRedirect(urlRedirect);
         response.setAuthOutcome(authorizationOutcome);
         response.setChannel(entity.getClientId());
-        response.setRequestId(entity.getGuid());
+        response.setRequestId(guid);
         response.setCorrelationId(entity.getCorrelationId());
         response.setIsOnboarding(entity.getIsOnboarding());
+        if (ObjectUtils.isNotEmpty(status)) {
+            try {
+                response.setPaymentRequestStatus(getEnumValueFromString(status));
+            } catch (IllegalArgumentException iae) {
+                String errorMsg = String.format("Invalid status %s for request entity with GUID: %s", status, guid);
+                log.error(errorMsg);
+                response.setError(errorMsg);
```

Comment:
@andrearovere I changed the response with a more self-explaining message.
You were using "No authorization outcome has been received yet" but it is not always true